### DISCUSSION TITLE
Add Custom Text on Single Day (v8)

### DISCUSSION THREAD

**Discussion by brennondenny** [gpbl/react-day-picker#1418] [Support] [answered]
I'm trying to add custom text on certain days, in the example below I'm using a custom modifier to add a border to a single set date (28th). How would I add text above the day? Without typescript

<img src="https://user-images.githubusercontent.com/17560356/160725859-3a9dfd6d-27f1-4e8e-9dad-5299337f55e7.png" width="250">

**Comment by gpbl** [comment, marked as answer]
@brennondenny have a look to this example:
https://codesandbox.io/s/sandpack-project-forked-huni2h

You can use the a `Day` custom component to add a content above the DayContent.

```jsx
import React from "react";
import { Day, DayProps, DayPicker, useActiveModifiers } from "react-day-picker";

function DayWithText(props) {
  const activeModifiers = useActiveModifiers(props.date, props.displayMonth);
  return (
    <span>
      <Day {...props} />
      {activeModifiers.today && "Today!"}
    </span>
  );
}

export default function App() {
  return <DayPicker components={{ Day: DayWithText }} />;
}
```

**Reply by ammarjmhor10** [reply]
Thanks for this answer. But how to pass a dynamic date like props ?

**Reply by gpbl** [reply]
@ammarjmhor10 you can't add extra props to the day component,  however you can create the component when the dynamic date is in the scope:

```tsx
export default function App(props: { dynamicDate: Date }) {
  const { dynamicDate = new Date() } = props;

  // The custom  component has now access to the "dynamic" date
  const DayWithText = (dayProps: DayProps) => {
    return (
      <span>
        <Day {...dayProps} />
        {isSameDay(dayProps.date, dynamicDate) && "Same day!"}
      </span>
    );
  };

  return (
    <DayPicker
      components={{
        Day: DayWithText
      }}
    />
  );
}
```

Example: https://codesandbox.io/s/sandpack-project-forked-0emlow?file=/App.tsx

**Comment by brennondenny** [comment]
@gpbl thanks but I'm having trouble adding a custom modifier for a specific date. I think I need to pass in arrayMatcher? 
In the example I'm trying to add this text to the 28th.

**Reply by gpbl** [reply]
@brennondenny would help to see you code. Start from the CodeSandbox i posted :)

**Reply by brennondenny** [reply]
Was able to get it, thanks
